bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyDiscoveryParticipant.java

```diff
@@ -70,89 +99,85 @@ public String getServiceType() {
+            // Get device settings
+            Configuration serviceConfig = configurationAdmin.getConfiguration("binding.shelly");
+            if (serviceConfig.getProperties() != null) {
+                bindingConfig.updateFromProperties(serviceConfig.getProperties());
             }
 
-            // Get device settings
-            Validate.notNull(bindingConfig);
+            ShellyThingConfiguration config = new ShellyThingConfiguration();
             config.deviceIp = address;
             config.userId = bindingConfig.defaultUserId;
             config.password = bindingConfig.defaultPassword;
-            ShellyHttpApi api = new ShellyHttpApi(config);
 
             try {
+                ShellyHttpApi api = new ShellyHttpApi(name, config, httpClient);
+
                 profile = api.getDeviceProfile(thingType);
-                Validate.notNull(profile);
-                logger.debug("Shelly settings : {}", profile.settingsJson);
-                Validate.notNull(profile, "Unable to get device profile: ");
-                model = profile.settings.device.type;
+                logger.debug("{}: Shelly settings : {}", name, profile.settingsJson);
+                model = getString(profile.settings.device.type);
                 mode = profile.mode;
 
                 properties = ShellyBaseHandler.fillDeviceProperties(profile);
-                logger.trace("name={}, thingType={}, deviceType={}, mode={}", name, thingType, profile.deviceType,
+                logger.trace("{}: thingType={}, deviceType={}, mode={}", name, thingType, profile.deviceType,
                         mode.isEmpty() ? "<standard>" : mode);
 
                 // get thing type from device name
-                thingUID = ShellyThingCreator.getThingUID(name, mode, false);
-            } catch (IOException e) {
-                if (e.getMessage().contains(APIERR_HTTP_401_UNAUTHORIZED)) {
-                    logger.warn("Device {} ({}) reported 'Access defined' (userid/password mismatch).", name, address);
+                thingUID = ShellyDeviceProfile.getThingUID(name, mode, false);
+            } catch (ShellyApiException e) {
+                ShellyApiResult result = e.getApiResult();
+                if (result.isHttpAccessUnauthorized()) {
+                    logger.info("{}: {}", name, messages.get("discovery.protected", address));
 
                     // create shellyunknown thing - will be changed during thing initialization with valid credentials
-                    thingUID = ShellyThingCreator.getThingUID(name, mode, true);
+                    thingUID = ShellyDeviceProfile.getThingUID(name, mode, true);
                 } else {
-                    logger.warn("Device discovery failed for device {}, IP {}: {} ({})\n{}", name, address,
-                            e.getMessage(), e.getClass(), e.getStackTrace());
+                    logger.info("{}: {}", name, messages.get("discovery.failed", address));
```

Comment:
info?

Reply:
yes, I want to inform the user. Mostly this requires the config of the credentials

Reply:
I removed the .info on an run-time exception and kept it for an API exception (e.g. unauthorized)